content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,148 @@
+     value:
+       name: istio.basic_auth
+       config_discovery:
+         config_source:
+           ads: {}
+           initial_fetch_timeout: 0s # wait indefinitely to prevent bad Wasm fetch
+         type_urls: [ "type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm"]
+{{< /text >}}
+
+The second `EnvoyFilter` resource provides configuration for the filter, which is composed as an `EXTENSION_CONFIG` patch and will be distributed to the proxy as an Envoy [`Extension Configuration`](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS) resource.
+Once this update reaches Istio agent, it will download the Wasm module file and store it at the local file system.
+If the download fails, Istio-agent will reject the `Extension Configuration` update and prevent bad Wasm filter configuration from reaching Envoy.
+Most of this `EnvoyFilter` configuration is boilerplate. The important parts are:
+
+- Wasm `vm` configuration which points to a remote Wasm module.
+- Wasm extension configuration, which is a Json string and consumed by the Wasm extension.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+ name: basic-auth-config
+ namespace: istio-system
+spec:
+ configPatches:
+ - applyTo: EXTENSION_CONFIG
+   match:
+     context: GATEWAY
+   patch:
+     operation: ADD
+     value:
+       name: istio.basic_auth
+       typed_config:
+         "@type": type.googleapis.com/udpa.type.v1.TypedStruct
+         type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
+         value:
+           config:
+             vm_config:
+               vm_id: basic-auth
+               runtime: envoy.wasm.runtime.v8
+               code:
+                 remote:
+                   http_uri:
+                     uri: https://github.com/istio-ecosystem/wasm-extensions/releases/download/1.9.0/basic-auth.wasm
+                   # Optional: specifying checksum will let istio agent
+                   # verify the checksum of download artifacts. Missing
+                   # checksum will cause the Wasm module to be downloaded
+                   # repeatedly
+                   sha256: 6b0cecad751940eeedd68de5b9bcf940d0aac8bfc5f61c18d71985ee9460ee77
+             # The configuration for the Wasm extension itself
+             configuration:
+               '@type': type.googleapis.com/google.protobuf.StringValue
+               value: |
+                 {
+                   "basic_auth_rules": [
+                     {
+                       "prefix": "/productpage",
+                       "request_methods":[ "GET", "POST" ],
+                       "credentials":[ "ok:test", "YWRtaW4zOmFkbWluMw==" ]
+                     }
+                   ]
+                 }
+{{< /text >}}
+
+{{< tip >}}
+Note: The basic auth Wasm module release follows Istio release version. In this example, the module is versioned and tested with Istio 1.9.
```

Comment:
I don't understand what this is trying to tell the user, please rephrase.

Reply:
Removed. I didn't know there is a way to version the yaml file in the doc so wanted to emphasize that this is only tested with Istio 1.9 release. But later I find out there is actually a way to version the EnvoyFilter with `{{< istio_version >}}` substitution, so I don't think we need this note any more.